test/core/xds/xds_transport_fake.h

```diff
@@ -66,6 +68,8 @@ class FakeXdsTransportFactory : public XdsTransportFactory {
 
     void Orphan() override;
 
+    void StartRecvMessage() override { read_count_ += 1; }
```

Comment:
I think we probably want something a little more sophisticated here:
- We should ensure that there is only one pending read at a time.  A read is considered pending between when `StartRecvMessage()` is called and when `EventHandler::OnRecvMessage()` is invoked.
- We should never invoke `EventHandler::OnRecvMessage()` without first seeing `StartRecvMessage()`.  This means that if the test calls `SendMessageToClient()` but `XdsClient` has not yet called `StartRecvMessage()`, the message should stay queued until `XdsClient` does call `StartRecvMessage()`.
- We probably want to expose a method for the test to check whether there is a pending read, so that we can write a test that shows that `XdsClient` does not start a new read until all watcher notifications have been completed.

Reply:
Done